### PR TITLE
[#2] Parse .cabal file for default extensions per stanza

### LINKED ISSUE #2: Parse .cabal file for default extensions per stanza
Could return the map of stanzas' directory to list of enabled extensions

### PR DESCRIPTION
**Pull request by chshersh**
Resolves #2

It almost works!

```haskell
λ: parseCabalExtensions "extensions.cabal"
fromList
    [
        ( "src/Extensions.hs"
        ,
            [ ConstraintKinds
            , DeriveGeneric
            , DerivingStrategies
            , GeneralizedNewtypeDeriving
            , InstanceSigs
            , KindSignatures
            , LambdaCase
            , OverloadedStrings
            , RecordWildCards
            , ScopedTypeVariables
            , StandaloneDeriving
            , TupleSections
            , TypeApplications
            , ViewPatterns
            ]
        )
    ,
        ( "src/Extensions/Cabal.hs"
        ,
            [ ConstraintKinds
            , DeriveGeneric
            , DerivingStrategies
            , GeneralizedNewtypeDeriving
            , InstanceSigs
            , KindSignatures
            , LambdaCase
            , OverloadedStrings
            , RecordWildCards
            , ScopedTypeVariables
            , StandaloneDeriving
            , TupleSections
            , TypeApplications
            , ViewPatterns
            ]
        )
    ]
```

### CODE REVIEW

File: src/Extensions/Cabal.hs [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+    Cabal.OverlappingInstances       -> Just OverlappingInstances
+    Cabal.UndecidableInstances       -> Just UndecidableInstances
+    Cabal.IncoherentInstances        -> Just IncoherentInstances
+    Cabal.DoRec                      -> Just RecursiveDo
+    Cabal.RecursiveDo                -> Just RecursiveDo
+    Cabal.ParallelListComp           -> Just ParallelListComp
+    Cabal.MultiParamTypeClasses      -> Just MultiParamTypeClasses
+    Cabal.MonomorphismRestriction    -> Just MonomorphismRestriction
+    Cabal.FunctionalDependencies     -> Just FunctionalDependencies
+    Cabal.Rank2Types                 -> Just RankNTypes
```
Comment (vrom911) [inline src/Extensions/Cabal.hs:139]: :cry:  My life is a lie

File: src/Extensions/Cabal.hs [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+    Cabal.StandaloneDeriving         -> Just StandaloneDeriving
+    Cabal.UnicodeSyntax              -> Just UnicodeSyntax
+    Cabal.UnliftedFFITypes           -> Just UnliftedFFITypes
+    Cabal.InterruptibleFFI           -> Just InterruptibleFFI
+    Cabal.CApiFFI                    -> Just CApiFFI
+    Cabal.LiberalTypeSynonyms        -> Just LiberalTypeSynonyms
+    Cabal.TypeOperators              -> Just TypeOperators
+    Cabal.RecordWildCards            -> Just RecordWildCards
+    Cabal.RecordPuns                 -> Just NamedWildCards
+    Cabal.NamedFieldPuns             -> Just NamedWildCards
```
Comment (vrom911) [inline src/Extensions/Cabal.hs:171]: :exploding_head:

File: src/Extensions/Cabal.hs [hunk 3]
```diff
@@ -0,0 +1,265 @@
...
+    Cabal.RestrictedTypeSynonyms -> Nothing
+    Cabal.HereDocuments          -> Nothing
+    Cabal.MonoPatBinds           -> Nothing
+    Cabal.XmlSyntax              -> Nothing
+    Cabal.RegularPatterns        -> Nothing
+    Cabal.SafeImports            -> Nothing
+    Cabal.Safe                   -> Nothing
+    Cabal.Trustworthy            -> Nothing
+    Cabal.Unsafe                 -> Nothing
+    Cabal.NewQualifiedOperators  -> Nothing
```
Comment (vrom911) [inline src/Extensions/Cabal.hs:265]: WOW :open_mouth:

**Review by vrom911 — APPROVED** [review]
This was a very informative reading! I learned a lot from this PR. Thank you :clap:

File: src/Extensions/Cabal.hs (hunk 3 above)
Comment (chshersh) [reply src/Extensions/Cabal.hs:265]: `XmlSyntax` is my favourite :rofl:

**Pull request merged**